GitFyle.Core.Api/Services/Foundations/Contributions/ContributionService.Exceptions.cs

```diff
@@ -0,0 +1,133 @@
+            }
+            catch (NullContributionException nullContributionException)
+            {
+                throw await CreateAndLogValidationExceptionAsync(nullContributionException);
+            }
+            catch (InvalidContributionException invalidContributionException)
+            {
+                throw await CreateAndLogValidationExceptionAsync(invalidContributionException);
+            }
+            catch (SqlException sqlException)
+            {
+                var failedStorageContributionException = new FailedStorageContributionException(
+                    message: "Failed contribution storage error occurred, contact support.",
+                    innerException: sqlException);
+
+                throw await CreateAndLogCriticalDependencyExceptionAsync(failedStorageContributionException);
+            }
+            catch (DuplicateKeyException duplicateKeyException)
+            {
+                var alreadyExistsContributionException =
+                    new AlreadyExistsContributionException(
+                        message: "Contribution already exists error occurred.",
+                        innerException: duplicateKeyException,
+                        data: duplicateKeyException.Data);
+
+                throw await CreateAndLogDependencyValidationExceptionAsync(alreadyExistsContributionException);
+            }
+            catch (DbUpdateException dbUpdateException)
+            {
+                var failedOperationContributionException =
+                    new FailedOperationContributionException(
+                        message: "Failed operation contribution  error occurred, contact support.",
+                        innerException: dbUpdateException);
+
+                throw await CreateAndLogDependencyExceptionAsync(failedOperationContributionException);
+            }
+            catch (Exception exception)
+            {
+                var failedServiceContributionException =
+                    new FailedServiceContributionException(
+                        message: "Failed service contribution error occurred, contact support.",
+                        innerException: exception);
+
+                throw await CreateAndLogServiceExceptionAsync(failedServiceContributionException);
+            }
+        }
+
+        private async ValueTask<ContributionValidationException> CreateAndLogValidationExceptionAsync(
+            Xeption exception)
+        {
+            var contributionValidationException = new ContributionValidationException(
+                message: "Contribution validation error occurred, fix errors and try again.",
+                innerException: exception);
+
+            await this.loggingBroker.LogErrorAsync(contributionValidationException);
+
+            return contributionValidationException;
+        }
+
+        private async ValueTask<ContributionDependencyException> CreateAndLogCriticalDependencyExceptionAsync(
+        Xeption exception)
```

Comment:
Indent when you wrap

```suggestion
        private async ValueTask<ContributionDependencyException> CreateAndLogCriticalDependencyExceptionAsync(
            Xeption exception)
```